addons/project/models/project_update.py

```diff
@@ -0,0 +1,253 @@
+             '|', ('deadline', '<', fields.Date.context_today(self) + relativedelta(years=1)), ('deadline', '=', False)])._get_data_list()
+        updated_milestones = self._get_last_updated_milestone(project_id=project_id)
+        created_milestones = Milestone.search(
+            [('project_id', '=', project_id),
+             ('create_date', '>', fields.Datetime.now() + timedelta(days=-30))])._get_data_list()
+        return {
+            'show_section': (list_milestones or updated_milestones or created_milestones) and True or False,
+            'list': list_milestones,
+            'updated': updated_milestones,
+            'created': created_milestones,
+        }
+
+    @api.model
+    def _get_last_updated_milestone(self, project_id):
+        Milestone = self.env['project.milestone']
+        query = """
+            SELECT DISTINCT pm.id as milestone_id,
+                            pm.deadline as deadline,
+                            FIRST_VALUE(old_value_datetime::date) OVER w_partition as old_value,
+                            pm.deadline as new_value
+                       FROM mail_message mm
+                 INNER JOIN mail_tracking_value mtv
+                         ON mm.id = mtv.mail_message_id
+                 INNER JOIN ir_model_fields imf
+                         ON mtv.field = imf.id
+                        AND imf.model = 'project.milestone'
+                        AND imf.name = 'deadline'
+                 INNER JOIN project_milestone pm
+                         ON mm.res_id = pm.id
+                      WHERE mm.model = 'project.milestone'
+                        AND mm.message_type = 'notification'
+                        AND pm.project_id = %(project_id)s
+                        AND mm.date > (now() at time zone 'utc')::date - '1 month'::interval
+                     WINDOW w_partition AS (
+                             PARTITION BY pm.id
+                             ORDER BY mm.date ASC
+                            )
+                   ORDER BY pm.deadline ASC;
+        """
+        self.env.cr.execute(
+            query,
+            {'project_id': project_id}
+        )
+        results = self.env.cr.dictfetchall()
+        return [{
+            **Milestone.browse(result['milestone_id'])._get_data(),
+            'new_value': result['new_value'],
+            'old_value': result['old_value'],
+        } for result in results]
+
+    def action_burndown_open_tasks(self):
+        action = self.env['ir.actions.act_window']._for_xml_id('project.action_project_task_burndown_chart_report')
+        project_id = self.project_id.id or self.env.context.get('default_project_id') or self.env.context.get('active_id')
+        context = {
+            'active_id': project_id,
+            'search_default_project_id': project_id,
+            'graph_mode': 'bar',
+        }
+        return dict(action, context=context)
+
+class ProjectMilestone(models.Model):
```

Comment:
You could split the different models into specific files.

Reply:
Done, thanks ! :)